### ISSUE TITLE
'override' auto-scrolling

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#2236]
sure most has done this:
A track is playing
you want to inspect something in the track, perhaps it was just something you heard, or diddent :p 
You mouse-grab the slider, and drag it, but to no avail 
Auto-scroll forces the focus to be at play-head.
What i thought was that is would be reasonable that any manual drag of the slider, would indicate that user want to 'over-ride' the current view-position, and let the view change even if auto-scroll was on. 

> When user then release the slider, the view would jump to auto-scroll pos. again.

NO i have done re-thinking on this. The reason the user want to move the view, is that he actually like to focus a _different_, than part, where the playhead is at _that_ moment. This fact makes this feature very simple to implement. The correct behavior is to simply **stop** playback, and then the user can freely scroll the view to any position of his choice. So the only thing that is needed is that user-scrolling of the view, should stop playback! Its as easy as that.

This feature should be for all editor-windows (curent lmms only have play-head-controlled scroll in P-R and S-E, but Automation-editor should hopefully soon(ish) be scrolling too)

**Comment by tresf** [comment]
:+1:

**Comment by StakeoutPunch** [comment]
I think this might be somewhat related so posting here (let me know if it isnt):

When you start or stop playback when in the piano roll, the view automatically cuts to bar 1 where the playhead is even with auto-scroll off.

**Comment by tresf** [comment]
> When you start or stop playback when in the piano roll, the view automatically cuts to bar 1 where the playhead is even with auto-scroll off.

@StakeoutPunch, I believe there's a bug report for what you are mentioning here:  #1735

**Comment by tresf** [comment]
Closing as duplicate of meta-bug **Better Playhead Auto-Scroll #2291**.  Please feel free to comment here if needed.

**Comment by regulus79** [comment]
If auto-scroll disables itself when the user drags the scrollbar, would it re-enable itself when the user presses play again?

**Comment by messmerd** [comment]
> If auto-scroll disables itself when the user drags the scrollbar, would it re-enable itself when the user presses play again?

Auto-scroll should not be disabled when the user is dragging the scrollbar - just overridden. So when the user releases their mouse to quit dragging the scrollbar, auto-scroll (if enabled) will snap the view back to where the play-head is.

With the current behavior, auto-scroll and the user's manual scrollbar interaction both fight for control, producing a UI bug. To fix it, we need to make the user interaction take precedence.

A similar issue occurs when dragging a knob while it's being automated or controlled. Ideally, the user's manual interaction should temporarily override any automation or controller, but it doesn't.

**Comment by musikBear** [comment]
>  we need to make the user interaction take precedence

Yes, but is the most simple to just stop playback at the current position User can continue playback with just a press on spacebar, either wehere the playhead currently is parked, or in a new position after user has moved the playhead manually
